pkg/danm/danm_test.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+  "testing"
+)
+
+var devicePool0 = "pool0"
+var devicePool1 = "pool1"
+
+func TestPopDevice(t *testing.T) {
+  allocatedDevices := make(map[string]*[]string)
+
+  device,err := popDevice(devicePool0, allocatedDevices)
+  if err == nil {
+    t.Errorf("Uninitialized map should expect error.")
+  }
+
+  allocatedDevices[devicePool0] = &[]string{"device0"}
+  allocatedDevices[devicePool1] = &[]string{"device1", "device2"}
+
+  device,err = popDevice(devicePool1, allocatedDevices)
+  if device != "device2"  || err != nil {
+    t.Errorf("Received device or error does not match with expectation.")
+  }
+  device,err = popDevice(devicePool1, allocatedDevices)
+  if device != "device1"  || err != nil {
+    t.Errorf("Received device or error does not match with expectation.")
+  }
+  device,err = popDevice(devicePool1, allocatedDevices)
+  if device == ""  && err == nil {
```

Comment:
in erroneous cases it is kind of standard to not expect anything regarding the value of the "other" return parameter
checking for error is enough